File: src/main/kotlin/de/rewe/kafkatogcs/schema/conversion/AvroConstants.java
package de.rewe.kafkatogcs.schema.conversion;

import tech.allegro.schema.json2avro.converter.JsonAvroConverter;

/**
 * {@link AvroConstants} holds constant refs to an {@link AvroNameTransformer}
 * and a {@link JsonAvroConverter}.
 */
public class AvroConstants {

  private AvroConstants() {}

  public static final AvroNameTransformer NAME_TRANSFORMER = new AvroNameTransformer();
  public static final JsonAvroConverter JSON_CONVERTER = JsonAvroConverter.builder()
      .setNameTransformer(NAME_TRANSFORMER::getIdentifier)
      .build();

}

File: src/main/kotlin/de/rewe/kafkatogcs/schema/conversion/AvroConstants.kt
package de.rewe.kafkatogcs.schema.conversion

import tech.allegro.schema.json2avro.converter.JsonAvroConverter

/**
 * [AvroConstants] holds constant refs to an [AvroNameTransformer]
 * and a [JsonAvroConverter].
 */
object AvroConstants {
    val NAME_TRANSFORMER = AvroNameTransformer()
    val JSON_CONVERTER = JsonAvroConverter.builder()
        .setNameTransformer { name: String? -> NAME_TRANSFORMER.getIdentifier(name) }
        .build()
}
